File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.py
def solution(numbers):
    a = len(str(max(numbers)))
    sortedNumber = sorted(numbers, key = lambda x:str(x)*a, reverse=True)

    answer = ''
    for i, n in enumerate(sortedNumber):
        if i == 0 and n == 0:
            answer = str(n)
            break
        else: answer = answer+str(n)

    return answer

File: 프로그래머스/2/42746. 가장 큰 수/가장 큰 수.js
function solution(numbers) {
    var answer = '';
    let a = String(Math.max(numbers)).length
    const sortedNumbers = numbers.sort((x, y) => {
        const sx = String(x).repeat(a);
        const sy = String(y).repeat(a);
        return sy.localeCompare(sx); // 내림차순
    });
    //console.log(sortedNumbers)
    for(const [idx, val] of sortedNumbers.entries()){
        if (val === 0 && idx === 0){
            answer += "0"
            break
        }
        else{
            answer += String(val)
        }
    }
    return answer;
}